core/iwasm/common/wasm_exec_env.c

```diff
@@ -282,8 +282,11 @@ wasm_exec_env_set_thread_info(WASMExecEnv *exec_env)
     os_mutex_lock(&exec_env->wait_lock);
 #endif
     exec_env->handle = os_self_thread();
-    exec_env->native_stack_boundary =
-        stack_boundary ? stack_boundary + WASM_STACK_GUARD_SIZE : NULL;
+    if (exec_env->user_native_stack_boundary)
+        exec_env->native_stack_boundary = exec_env->user_native_stack_boundary;
```

Comment:
i suspect it's simpler to add WASM_STACK_GUARD_SIZE here.
or, at least document it prominently on the api doc of wasm_runtime_set_native_stack_boundary to ensure the user aware of it.

Reply:
Yes, seems it is simpler to add it. I just thought the default value of WASM_STACK_GUARD_SIZE depends on platforms, and it may be more flexible for developer to decide what the boundary should be. Both are good to me, and agree to add comment about WASM_STACK_GUARD_SIZE in the API, event when WASM_STACK_GUARD_SIZE is added here.

Reply:
> I still think that it should be better for developer to choose what to do: if the behavior controlled by runtime, then all the two options are controlled by runtime, otherwise all the two options had better be controlled by developer, or it may be confusing for him. I updated the PR to add some comments, please help review again. Thanks.

Agree, since it's an API to give the control to the user, then we should leave everything to the user.